### PR TITLE
VIV-33517 User data persistence with Airtable service

### PR DESCRIPTION
**Pull request by sarahmolinerroy**

### CODE REVIEW

File: user-data-persistence/code/lib/services/airtable.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    const url = properties.get("config", "baseUrl") + "/" + dbUserId
+    const options = {
+      format: "json",
+      cacheTime: 0,
+      headers: { 
+        Authorization: "Bearer " + properties.get("secret", "apiKey")
+      }
+    }
+    const response = http.deleteUrl(url, {}, options)
+    if (response) {
```
Comment (siamakh) [inline user-data-persistence/code/lib/services/airtable.js]: ignorable comment :) 
I like being clear on what is the return value but maybe a ternary operation is as clear and more concise? 
`return response ? true : false`

File: user-data-persistence/code/lib/services/airtable.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  }
+  const response = http.getUrl(url, options)
+  if (response && response.records && response.records.length === 1) {
+    const userData = JSON.parse(response.records[0].fields[properties.get("config", "userDataField")])
+    userData.$id = response.records[0].id
+    return userData
+  } else {
+    // Doesn't exist
+    return
```
Comment (siamakh) [inline user-data-persistence/code/lib/services/airtable.js]: return null

File: user-data-persistence/README.md [hunk 3]
```diff
@@ -30,36 +31,25 @@ This will delete the user data. `nl("Disband my army")`
...
-    - `userIdField`=`bixby-user-id`: User id field from above
-    - `userDataField`=`bixby-user-data`: User data field from above
-  - Secrets:
-    - `apiKey`=`678b16ce31cfef34e796dcefd81ea27072574`: Update this to match the Server apiKey provided in your Database Settings
-  - Hit the Save button. Now your Configs & Secrets should look like this: ![Configs & Secrets](./screenshots/configs_and_secrets.png)
-- Edit the capsule.bxb file to update the [`id` key](https://bixbydevelopers.com/dev/docs/reference/type/capsule.id) to match your registered capsule namespace and name ![Capsule ID](./screenshots/capsule_id.png)
-- Edit the capsule.properties file to change the `capsule.config.mode` from `example` to `default` ![Capsule Config Mode](./screenshots/capsule_config_mode.png)
+- [Add configs & secrets](https://bixbydevelopers.com/dev/docs/reference/ref-topics/capsule-config#config-secrets) in the Bixby Developer Center. Use the field names and values provided in the integration walk-throughs for your selected service. Hit the Save button. Now your Configs & Secrets should look like this: ![Configs & Secrets](./screenshots/configs_and_secrets.png)
```
Comment (jatinbviv) [inline user-data-persistence/README.md]:
```suggestion
- [Add configs & secrets](https://bixbydevelopers.com/dev/docs/reference/ref-topics/capsule-config#config-secrets) in the Bixby Developer Center. Use the field names and values provided in the integration walk-throughs for your selected service. Click the **Save** button. Now your Configs & Secrets should look like this: ![Configs & Secrets](./screenshots/configs_and_secrets.png)
```

File: user-data-persistence/airtable.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Airtable
+
+Airtable is a service which allows developers to create a cloud hosted spreadsheet exposed through a REST API.
```
Comment (jatinbviv) [inline user-data-persistence/airtable.md]:
```suggestion
Airtable is a service which allows developers to create a cloud-hosted spreadsheet exposed through a REST API.
```

File: user-data-persistence/airtable.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+To try it out using the provided test account, edit the `capsule.properties` file
+to say `capsule.config.mode=exampleAirtable`.
```
Comment (jatinbviv) [inline user-data-persistence/airtable.md]:
```suggestion
to include `capsule.config.mode=exampleAirtable`.
```

File: user-data-persistence/restDB.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+## RestDB
+
+RestDB is a service which allows developers to create a cloud hosted noSQL
+database exposed through a REST API.
+
+To try it out using the provided test account, edit the `capsule.properties` file
+to say `capsule.config.mode=exampleRestDB`.
```
Comment (jatinbviv) [inline user-data-persistence/restDB.md]:
```suggestion
to include `capsule.config.mode=exampleRestDB`.
```

File: user-data-persistence/restDB.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+#### Remote Database Setup
+
+- Sign up for a free restDB account at https://restdb.io/
+- [Create a new Database](https://restdb.io/account/databases/) called `bixby`. Click on the database to go to its home page. Example: https://bixby-25e7.restdb.io/home/ ![Database](./screenshots/restdb/database.png)
+- Click on the top right icon with 3 gears to enter Developer Mode. Add a new Collection called `<your-capsule-name>`. Example: `example-user-data` ![New Collection](./screenshots/restdb/new_collection.png)
+- Click on the new Collection to configure its Fields.
+  - Add Field for user id of type Text with Requirements Required and Unique. Example: `bixby-user-id` ![Id Field](./screenshots/restdb/id_field.png)
+  - Add Field for user data of type json. Example: `bixby-user-data` ![Data Field](./screenshots/restdb/data_field.png)
+- Your setup should now look like this and is ready to be connected to the capsule. ![Collection](./screenshots/restdb/collection.png)
+- Follow the instructions in the [main README](./README.md) and insert the
```
Comment (jatinbviv) [inline user-data-persistence/restDB.md]:
```suggestion
- Follow the instructions in the [main README](./README.md) and include the
```

**Pull request merged**